homeassistant/components/geo_location/nsw_rural_fire_service_feed.py

```diff
@@ -0,0 +1,246 @@
+            _LOGGER.debug("New entity added %s", new_entity)
+            new_entities.append(new_entity)
+        # Add new entities to HA and keep track of them in this manager.
+        self._add_entities(new_entities, True)
+        self._managed_entities.extend(new_entities)
+
+    def get_feed_entry(self, external_id):
+        """Return a feed entry identified by external id."""
+        return next((entry for entry in self._feed_entries
+                     if entry.external_id == external_id), None)
+
+
+class NswRuralFireServiceLocationEvent(GeoLocationEvent):
+    """This represents an external event with GeoJSON data."""
+
+    def __init__(self, feed_manager, feed_entry):
+        """Initialize entity with data from feed entry."""
+        self._feed_manager = feed_manager
+        self._update_from_feed(feed_entry)
+
+    @property
+    def should_poll(self):
+        """No polling needed for GeoJSON location events."""
+        return False
+
+    async def async_update(self):
+        """Update this entity from the data held in the feed manager."""
+        feed_entry = self._feed_manager.get_feed_entry(self.external_id)
+        if feed_entry:
+            self._update_from_feed(feed_entry)
+
+    def _update_from_feed(self, feed_entry):
+        """Update the internal state from the provided feed entry."""
+        self._name = feed_entry.title
+        self._distance = feed_entry.distance_to_home
+        self._latitude = feed_entry.coordinates[0]
+        self._longitude = feed_entry.coordinates[1]
+        self.external_id = feed_entry.external_id
+        self._attribution = feed_entry.attribution
+        self._category = feed_entry.category
+        self._publication_date = feed_entry.publication_date
+        self._location = feed_entry.location
+        self._council_area = feed_entry.council_area
+        self._status = feed_entry.status
+        self._type = feed_entry.type
+        self._fire = feed_entry.fire
+        self._size = feed_entry.size
+        self._responsible_agency = feed_entry.responsible_agency
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def distance(self) -> Optional[float]:
+        """Return distance value of this external event."""
+        return self._distance
+
+    @property
+    def latitude(self) -> Optional[float]:
+        """Return latitude value of this external event."""
+        return self._latitude
+
+    @property
+    def longitude(self) -> Optional[float]:
+        """Return longitude value of this external event."""
+        return self._longitude
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return DEFAULT_UNIT_OF_MEASUREMENT
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {}
+        if self.external_id:
```

Comment:
In case you feel like removing some repeating code:

```python
for key, value in (
    (ATTR_EXTERNAL_ID, self.external_id),
    (ATTR_CATEGORY, self. _category),
):
    if value:
        attributes[key] = value
```

Reply:
Sounds good, will do.